labellab-flask/api/controllers/issuecontroller.py

```diff
@@ -3,36 +3,136 @@
 from flask import make_response, request, jsonify, current_app
 from flask_jwt_extended import (
     jwt_required,
+    get_jwt_identity,
 )
 
 from api.config import config
 from api.models.Issue import Issue
 
+from api.helpers.issue import (
+    save as save_issue
+)
+from api.middleware.project_member_access import project_member_only
+
+allowed_priorities = config[os.getenv("FLASK_CONFIG") or "development"].ISSUE_PRIORITIES_ALLOWED
+allowed_statuses = config[os.getenv("FLASK_CONFIG") or "development"].ISSUE_STATUSES_ALLOWED
+allowed_categories = config[os.getenv("FLASK_CONFIG") or "development"].CATEGORIES_ALLOWED
+allowed_entity_types = config[os.getenv("FLASK_CONFIG") or "development"].ENTITY_TYPES_ALLOWED
+
 class CreateIssues(MethodView):
-    """
-    This class-based view handles fetching all issues within a project
-    Url --> /api/v1/issue/create/<int:project_id>
-    """
+    """This class creates a new Issue."""
     @jwt_required
+    @project_member_only
     def post(self, project_id):
+        """
+        Handle POST request for this view.
+        Url --> /api/v1/issue/create/<int:project_id>
+        """
+        # getting JSON data from request
+        post_data = request.get_json(silent=True, force=True)
+        current_user = get_jwt_identity()
+        # Load model with necessary fields 
         try:
-                        
+            title = post_data["title"]
+            description = post_data["description"]
+            category = post_data["category"]
+        except Exception:
             response = {
-                "success": True,
-                "msg": "Issue Created",
-                "body": "New Issue Create for this project"
+                "success": False,
+                "msg": "Please provide all the required fields."
             }
-            return make_response(jsonify(response)), 200
+            return make_response(jsonify(response)), 400
+        if category not in allowed_categories:
+                print("Error occured: category not allowed")
```

Comment:
Let's not have print statements in the new code. We have them in the code written earlier but let's avoid them now.